src/aatoolbox/datasources/codab/codab.py

```diff
@@ -41,13 +42,17 @@ class CodAB(DataSource):
 
     def __init__(self, country_config: CountryConfig):
         super().__init__(
-            country_config, datasource_base_dir="cod_ab", is_public=True
+            country_config,
+            datasource_base_dir="cod_ab",
+            is_public=True,
+            config_attribute_name="codab",
         )
         self._raw_filepath = (
             self._raw_base_dir / f"{self._country_config.iso3}_"
             f"{self._datasource_base_dir}.shp.zip"
         )
 
+    @typing.no_type_check
```

Comment:
I think otherwise it would produce an error because no object is returned despite the return type being specified as `Path`? Is that right?

Reply:
Uff so actually it's because since I now made `codab` optional in the config file, mypy complains about the `self._country_config.codab` parametmers since they could be `None`. In the past I always managed to solve it by checking in `__init__` that it it's not None, but for some reason it doesn't work here!